core/src/main/java/org/apache/calcite/sql/type/SqlTypeName.java

```diff
@@ -65,12 +65,16 @@ public enum SqlTypeName {
   DATE(PrecScale.NO_NO, false, Types.DATE, SqlTypeFamily.DATE),
   TIME(PrecScale.NO_NO | PrecScale.YES_NO, false, Types.TIME,
       SqlTypeFamily.TIME),
-  TIME_WITH_LOCAL_TIME_ZONE(PrecScale.NO_NO | PrecScale.YES_NO, false, Types.OTHER,
+  TIME_WITH_LOCAL_TIME_ZONE(PrecScale.NO_NO | PrecScale.YES_NO, false, Types.TIME,
+      SqlTypeFamily.TIME),
+  TIME_TZ(PrecScale.NO_NO | PrecScale.YES_NO, false, Types.TIME,
```

Comment:
Going through this PR I'm also not opposed to changing the local time zones to something like `TIMESTAMP_LTZ`, I think we actually use that in some parts of the code. Either way should probably be consistent. What do you think?

Reply:
That would be nice, but this may break any third party software that uses Calcite.
Since we can't remove deprecated functions I assume we can't rename public enum fields either.